src/common/utils/resolve_reference.py

```diff
@@ -124,9 +128,11 @@ class AttributeItem:
     path: str
 
     def __repr__(self):
-        return f'Attribute="{self.path}"'
+        return self.path
 
-    def get_child(self, document: dict | list, data_source: DataSource, get_data_source: Callable) -> tuple[Any, str]:
+    def get_child(
+        self, document: dict | list, data_source: DataSource, get_data_source: Callable, resolve: bool = False
```

Comment:
This function does not use the resolve input param. So the param only has effect on queries? That's unintuitive.
I don't like this new resolve parameter. I think resolve_reference never should return a resolved document. It should only resolve while nesting downwards. If the caller needs to get the entity resolved it can do so itself easily